[Fundamentals ] Audit Logs (#23784)
cloudflare/cloudflare-docs: src/content/docs/fundamentals/account/account-security/audit-logs.mdx

  badge: 
    text: Beta

---

import { APIRequest } from "~/components"

Cloudflare Audit Logs are account-based. All user-initiated actions are recorded automatically across both the Cloudflare API and Dashboard. System-initiated logs are also captured to reflect actions taken automatically by Cloudflare systems, such as configuration updates, background processes, or internal policy enforcement.

When a user-initiated action triggers additional automated behavior, corresponding system-initiated logs may be generated. In some cases, these system-initiated logs include additional enrichments that provide more context about what was changed, offering deeper visibility into the full lifecycle of the action.

When an action occurs, it is streamed through Cloudflare's audit logging pipeline and stored. This ensures consistent visibility into activity across all products.

For more detailed information about how the user-initiated actions are logged automatically, refer to the [Cloudflare Blog](https://blog.cloudflare.com/introducing-automatic-audit-logs/).

:::note
A transition plan from Audit Logs v1 to Audit Logs v2 will be communicated in due course. 
:::

## Key features

> Audit Logs (version 2) provide a unified and standardized system for tracking and recording actions across Cloudflare products. This system enhances transparency and accountability by offering comprehensive insights into user-initiated and system-initiated activities within your Cloudflare environment. 

Reviewer: ```suggestion
Audit Logs (version 2) provides a unified and standardized system for tracking and recording actions across Cloudflare products. This system enhances transparency and accountability by offering comprehensive insights into user-initiated and system-initiated activities within your Cloudflare environment. 
```

- **Standardized logging**: Audit logs are automatically generated in a consistent format across all Cloudflare services, ensuring uniformity and eliminating inconsistencies.
- **Expanded product coverage**: Audit Logs covers 111 products, capturing actions from key endpoints, such as `/accounts`, `/zones`, `/user`, and `/memberships` APIs.
- **Granular filtering**: Uniformly formatted logs allow for precise filtering by actions, actors, methods, and resources, facilitating efficient investigations.

> - **Enhanced context and transparency**: Each log entry includes detailed context, such as the authentication method used, the interface (API or Dashboard) through which the action was performed, and mappings to Cloudflare Ray IDs for improved traceability.

Reviewer: ```suggestion
- **Enhanced context and transparency**: Each log entry includes detailed context, such as the authentication method used, the interface (API or dashboard) through which the action was performed, and mappings to Cloudflare Ray IDs for improved traceability.
```

- **Comprehensive activity capture**: Beyond create, edit, and delete actions, Audit Logs records GET requests and failed attempts, ensuring no critical activity is overlooked. 

## Retention

> Audit Logs are retained for 18 months before being deleted. Enterprise customers can use [Logpush](/logs/logpush/) to store Audit Logs for longer periods of time. Logpush for Audit Logs v2 will be available for GA. 

Reviewer: ```suggestion
Audit logs are retained for 18 months before being deleted. Enterprise customers can use [Logpush](/logs/logpush/) to store audit logs for longer periods of time. Logpush for Audit Logs v2 will be available for GA. 
```

## Access Audit Logs

> You can retrieve audit logs using one of the API or the dashboard. Audit Logs v2 will soon be available with Logpush. 

Reviewer: ```suggestion
You can retrieve audit logs using either the API or the dashboard. Audit Logs v2 will soon be available with Logpush. 
```

### API

> Audit Logs are available through the Cloudflare API. To retrieve audit logs, use the following endpoint:

Reviewer: ```suggestion
Audit logs are available through the Cloudflare API. To retrieve audit logs, use the following endpoint:
```

- `Before` (required) : End date for the audit log retrieval in the format yyyy-mm-dd.

```bash
GET https://api.cloudflare.com/client/v4/accounts/1234567890abcdef/logs/audit?since=2025-03-01T00:00:00Z&before=2025-03-26T23:59:59Z
```

```json title="Example response"
{
  "result": [
    {
      "action": "zone.settings.change",
      "actor": {
        "email": "user@example.com",
        "id": "0987654321abcdef"
      },
      "ip": "192.0.2.1",
      "method": "PUT",
      "interface": "dashboard",
      "resources": [
        {
          "resource_id": "zone123",
          "resource_type": "zone"
        }
      ],
      "timestamp": "2025-03-15T14:25:37Z"
    }
    // Additional log entries
  ],
  "success": true,
  "errors": [],
  "messages": []
}
```

For more information refer to the [API documentation](https://developers.cloudflare.com/api/resources/accounts/subresources/logs/subresources/audit/methods/list/#(params)%20default%20%3E%20(param)%20since%20%3E%20(schema)). 

### Dashboard

To access audit logs in the Cloudflare dashboard:

1. Log in to the [Cloudflare dashboard](https://dash.cloudflare.com/login) and select your account.

> 2. Go to **Manage Account** > **Audit Log**.

Reviewer: ```suggestion
2. Go to **Manage Account** > **Audit Logs**.
```
